### PR TITLE
new multiPartition

### PR DESCRIPTION
**Pull request by sithmel**
Test on iterator disposal will be carried out after merging https://github.com/sithmel/iter-tools/issues/171

discussion on https://github.com/sithmel/iter-tools/issues/116

### CODE REVIEW

**Comment by sithmel** [comment]
I am pretty happy with this. I am going to merge it soon

**Comment by sithmel** [comment]
The async version has an issue. I am looking at it

File: src/async-multi-partition.mjs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  try {
+    while (true) {
+      const iter = part(queueNumber++)
+      // this ensure to call the finally clause of iter
+      iter.next()
+      yield iter
+    }
+  } finally {
+    // I trim the number of queues
+    queues.length = maxQueues = queueNumber
```
Comment (KSXGitHub) [inline src/async-multi-partition.mjs]: Maybe this is what causes [the issue that you talked about](https://github.com/sithmel/iter-tools/pull/199#issuecomment-466798162).

Comment (sithmel) [reply src/async-multi-partition.mjs]: no, the cause was very subtle. But it is now fixed https://github.com/sithmel/iter-tools/pull/199/commits/895572f3cd7486f0cf67b10ea6a26acbba86212d#diff-bc98b2acad88eb39be94dcd71e05d22bR27

**Pull request merged**